src/o365/spo/commands/web/web-set.ts

```diff
@@ -102,6 +103,26 @@ class SpoWebSetCommand extends SpoCommand {
 
     request
       .patch(requestOptions)
+      .then((): Promise<void> => {
+        if (typeof args.options.welcomePage === 'undefined') {
+          return Promise.resolve();
+        }
+        if (this.verbose) {
+          cmd.log(`Updating Welcome page for the site ${args.options.webUrl}`);
+        }
+        const requestOptions: any = {
+          url: `${args.options.webUrl}/_api/web/rootfolder`,
+          headers: {
+            'Content-Type': 'application/json;odata=nometadata',
+            accept: 'application/json;odata=nometadata',
+            'IF-MATCH': '*',
+            'X-HTTP-Method': 'PATCH'
+          },
+          __metadata: {'type': 'SP.Folder'},
+          body: `{'WelcomePage':'${args.options.welcomePage}'}`
```

Comment:
Rather than manually escaping the payload, we can set it to a JS object and then on the request options set `json: true` and have the CLI handle escaping for us.